src/c_api/c_api.cc

```diff
@@ -946,5 +947,45 @@ XGB_DLL int XGBoosterSaveRabitCheckpoint(BoosterHandle handle) {
   API_END();
 }
 
+XGB_DLL int XGBoosterGetFeatureInteractions(BoosterHandle handle,
+                                            int max_fi_depth,
+                                            int max_tree_depth,
+                                            int max_deepening,
+                                            int ntrees,
+                                            const char *fmap_file,
+                                            xgboost::bst_ulong *out_len,
+                                            const char ***out_fi_array) {
+  API_BEGIN();
+  xgboost::FeatureMap fmap;
+  if (*fmap_file != '\0') {
+    try {
+      std::unique_ptr<dmlc::Stream> fs(dmlc::Stream::Create(fmap_file, "r"));
+      dmlc::istream is(fs.get());
+      fmap.LoadText(is);
+    } catch (...) {
+      LOG(CONSOLE) << "Warning: unable to read feature map: \"" << fmap_file << "\", "
+        "feature names wont be mapped";
+    }
+  }
+  Booster* bst = static_cast<Booster*>(handle);
+  xgbfi::XgbModelDump dump = bst->learner()->DumpModel(fmap, true, "text");
+  xgbfi::XgbModel model = xgbfi::XgbModelParser::GetXgbModelFromDump(dump, ntrees);
+  xgbfi::FeatureInteractions fi = model.GetFeatureInteractions(max_fi_depth,
+                                                               max_tree_depth,
+                                                               max_deepening);
```

Comment:
I think you don't need all of these classes here. Is it possible to have one single function (not method) GetFeatureInteractions() here? That way you do not need to define these classes in xgbfi.h, you can instead define them in the .cc file. This is cleaner as these classes are only specific to xgbfi and are not needed in other code.

Reply:
sure, will do.